File: C++/Medium/WrongAnswer/find-and-replace-in-string_7.cpp
class Solution {
public:
    string findReplaceString(string s, vector<int>& indices, vector<string>& sources, vector<string>& targets) {
        
        string res = "";
        int i = 0;
        int n = indices.size();
        int j = 0;
        while(i < s.size()) {
            if(j < n && i == indices[j]) {
                if(s.substr(i, sources[j].size()) == sources[j]) {
                    res += targets[j];
                    i += sources[j].size();
                } else {
                    res += s[i];
                    i++;
                }
                j++;
            } else {
                res += s[i];
                i++;
            }
        }
        return res;
    }
};

File: Python3/Medium/Accepted/find-and-replace-in-string_7.py
class Solution:
    def findReplaceString(self, s: str, indices: List[int], sources: List[str], targets: List[str]) -> str:
        
        s = list(s)
        for i, source, target in sorted(zip(indices, sources, targets), reverse=True):
            if all(i + j < len(s) and s[i + j] == c for j, c in enumerate(source)):
                s[i:i + len(source)] = target
        return "".join(s)